src/psyclone/psyir/tools/definition_use_chains.py

```diff
@@ -630,6 +630,13 @@ def _find_basic_blocks(self, nodelist):
                 # No control for the condition - we always check that.
                 control_flow_nodes.append(None)
                 basic_blocks.append([node.condition])
+                if node.ancestor((Loop, WhileLoop)):
+                    control_flow_nodes.append(node)
+                    basic_blocks.append(node.if_body.children[:])
+                    if node.else_body:
+                        control_flow_nodes.append(node)
+                        basic_blocks.append(node.else_body.children[:])
+                    continue
```

Comment:
@arporter I'm happy with this change - it is similar to my suggestion in #3061 but correctly handles whether there is an else_body.

@sergisiso There should probably be a unit test for this section added in `definition_use_chains_forward_dependence_test.py::test_definition_use_chain_find_basic_blocks` though, as the code inside that doesn't currently consider this case.